### DISCUSSION TITLE
Android error: invalid source release: 17

### DISCUSSION THREAD

**Discussion by lpickford-pixelbeard** [codemagic-ci-cd/codemagic-docs#2281] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Create an android project with the following set in the `app/build.gradle` `android { ... }` block:
```
android {
    ...

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_17
        targetCompatibility JavaVersion.VERSION_17
    }

    kotlinOptions {
        jvmTarget = "17"
    }

    ...
```
2. Build using codemagic

Other issues solved this by downgrading a library they were using, we are not able to do this as the version of a library we use is required by us.

### Expected results

The app builds

### Actual results

The following error is thrown during the build process:
```
* What went wrong:
Execution failed for task ':stripe_android:compileReleaseJavaWithJavac'.
> error: invalid source release: 17
```

### Build id (optional)

65806eb3b46a829428df61b8

**Comment by dtrdic** [comment, marked as answer]
Hello @lpickford-pixelbeard, based on the error message, it appears that Java 17 is expected in the build, but an incorrect version of Java is currently being used. If not specified, the default version is Java 11. Could you please try defining the Java version explicitly in your YAML file as shown below

```
    test-workflow:
      name: iOS Workflow
      instance_type: mac_mini_m1
      max_build_duration: 120
      environment:
        java: 17
```

**Reply by lpickford-pixelbeard** [reply]
Setting an environment variable solved it, found the correct one here: https://docs.codemagic.io/specs/versions-macos/
